proposals/0026/README.md

```diff
@@ -0,0 +1,275 @@
+As mentioned previously, backup and restoration commands will be added to restore from Kubernetes API unavailability and could help support switching to a different underlying Kubernetes cluster should the primary cluster become unavailable.
+
+### Tradeoffs
+
+The biggest architectural tradeoff in implementing this proposal is ease of feature development for simplicity of administration.
+Kubernetes has a lot of industry hype, baggage (but not literal [baggage][k8s-baggage]), and [public failure stories][k8s-af].On-premise operators may not have experience with using or operating Kubernetes and be wary of adding additional complexity.
+Operators have numerous options to use and deploy Kubernetes, so in order to mitigate concerns around complexity, we will provide extensive documentation on different alternatives to meet an operator's objectives.
+
+Examples include:
+* A cloud provider managed Kubernetes solution (GKE, AKS, EKS)
+* A single host [K3s][k3s] cluster
+* A multi-host on-premise cluster
+
+<!-- Link found when Google searching "Kubernetes baggage" -->
+[k8s-baggage]: https://www.redbubble.com/shop/kubernetes+bags
+[k8s-af]: https://k8s.af/
+[kcp]: https://github.com/kcp-dev/kcp/
+[k3s]: https://k3s.io
+
+### User Experience
+
+In order to use Tinkerbell, clients would interact with the Kubernetes API.
+In order to provision a machine, the steps would be:
+
+1. User creates a Hardware CRD object in Kubernetes.
+   This is analogous to the current `tink hardware push < hardware.json` command.
+1. User creates a Template CRD object in Kubernetes.
+   This is analogous to the current `tink template create < template.json` command
+1. User creates a Workflow CRD object in Kubernetes.
+   All the user would need to specify in the object `.spec` would be a reference to the Template object, and mappings for devices.
+   This is analogous to the current `tink workflow create -t $TEMPLATE_ID -r "{\"device_1\":\"$MAC_ADDR\"}` command.
+1. The Tinkerbell API would include a Kubernetes workflow controller that would watch for created workflows, and fill out the `.status` with most of the logic that currently exists in the [`CreateWorkflow()`][createwfrpc] RPC.
+1. The Tinkerbell API could subscribe to Workflow CRD changes, and stream them on to Tinkerbell worker clients over the existing [`GetWorkflowContexts()`][getwfctxs] streaming RPC.
+1. Tinkerbell worker clients would continue to call the existing Tinkerbell APIs to execute workflows.
+  The Tinkerbell API would store updates in a Workflow CRD `.status`
+
+[createwfrpc]: https://github.com/tinkerbell/tink/blob/b217be8/grpc-server/workflow.go#L19-L72
+[getwfctxs]: https://github.com/tinkerbell/tink/blob/a56e5cf9/protos/workflow/workflow.proto#L75
+
+## System-context-diagram
+
+See above diagrams.
+
+## APIs
+
+The following [Tinkerbell Workflow APIs][wf-apis] are used by the Tink worker, and would remain.
```

Comment:
Why can these services not be initially replaced by the Kubernetes model? Do you see them eventually be removed?

Do status conditions, events, logs, and additional CRDs offer the means to move these entities to the Kubernetes API?

.. I think I see.

These are the APIs needed for the provisioning machine to get actions and update state.  That could be more clearly defined.

Why would Tink workers not become Kubernetes clients, for example? Each machine could use a unique service account which would open up more possibilities (potentially reducing the machine threat model).